tests/gis_tests/geoapp/test_functions.py

```diff
@@ -223,7 +223,7 @@ def test_intersection(self):
         geom = Point(5, 23, srid=4326)
         qs = Country.objects.annotate(inter=functions.Intersection('mpoly', geom))
         for c in qs:
-            if spatialite or mysql or oracle:
+            if spatialite or mysql and not connection.ops.returns_invalid_geometry_collection or oracle:
```

Comment:
I'd add parenthesis to make the grouping clear: `(mysql and not connection.ops.returns_invalid_geometry_collection)`